### PR TITLE
Adding graphpro pip package

### PR DESCRIPTION
**Pull request by pegerto**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by pegerto** [comment]
@conda-forge/help-python

File: recipes/graphpro/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    - pip check
+
+about:
+  home: https://github.com/pegerto/graphpro
+  summary: 'GraphPro: a python framework for protein deep geometrical learning'
+  description: |
+    GraphPro is a versatile and pluggable OO python library designed for leveraging deep graph learning representations to gain
+    insights into structural proteins and their conformations. With its wide range of functionalities,
+    we provide a comprehensive framework for researchers and practitioners to explore and analyze complex protein structures.
+  license: GPL3
```
Comment (marcelotrevisani) [inline recipes/graphpro/meta.yaml:38]:
```suggestion
  license: GPL-3.0
```

File: recipes/graphpro/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+
+requirements:
+  host:
+    - python
+    - pip
+    - hatchling >=1.13.0
+  run:
+    - python
```
Comment (marcelotrevisani) [inline recipes/graphpro/meta.yaml:23]:
```suggestion
    - python >=3.7
    - pip
    - hatchling >=1.13.0
  run:
    - python >=3.7
```

File: recipes/graphpro/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 376e6b948eedcb2c06eec6576dd389011f43a682449e99478504832ed8726e13
+
+
+build:
+  script: {{ PYTHON }} -m pip install . -vv
```
Comment (marcelotrevisani) [inline recipes/graphpro/meta.yaml:14]:
```suggestion
build:
  noarch: python
  script: {{ PYTHON }} -m pip install . -vv
```

File: recipes/graphpro/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+requirements:
+  host:
+    - python
+    - pip
+    - hatchling >=1.13.0
+  run:
+    - python
+
+test:
+  requires:
```
Comment (marcelotrevisani) [inline recipes/graphpro/meta.yaml:26]:
```suggestion
test:
  imports:
    - graphpro
  requires:
```

**Comment by pegerto** [comment]
Thanks for the review @marcelotrevisani

**Review by marcelotrevisani — APPROVED** [review]

**Pull request merged**